akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -105,6 +106,11 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
         ctx.log.warning("Failed to delete events to sequence number [{}] due to [{}].", toSequenceNr, failure)
     }
 
+    val journalResponses = anyCtx.messageAdapter[JournalProtocol.Response](InternalProtocol.JournalResponse.apply)
+    val snapshotResponses = anyCtx.messageAdapter[SnapshotProtocol.Response](InternalProtocol.SnapshotterResponse.apply)
+    val permitterResponses =
+      anyCtx.messageAdapter[RecoveryPermitter.RecoveryPermitGranted.type](_ => InternalProtocol.RecoveryPermitGranted)
```

Comment:
Using `messageAdapter` instead of `widen` works fine in EventSourcedBehaviorImpl. We were uncertain about what the problem was but whatever that was it seems fixed now. However, messageAdapter doesn't solve or improve much here since the converted `InternalProtocol` messages will anyway be pass the behavior stack of outer interceptors. A plain interceptor would be more efficient (but still has the same problem with outer interceptors).

Reply:
Ach, when thinking about this before I didn't think of the fact that here we have a user specified message type and need to combine that with our internal and completely separate message type here.

And the internal messages must pass the entire behavior stack because if not they'd not be covered by for example supervision when that is used.